pandas/core/generic.py

```diff
@@ -5291,47 +5288,162 @@ def replace(self, to_replace=None, value=None, inplace=False, limit=None,
+            Bernstein basis
+        scipy.interpolate.interp1d : interpolate a 1-D function
+        scipy.interpolate.KroghInterpolator : interpolate polynomial (Krogh
+            interpolator)
+        scipy.interpolate.PchipInterpolator : PCHIP 1-d monotonic cubic
+            interpolation
+        scipy.interpolate.CubicSpline : cubic spline data interpolator
+
+        Notes
+        -----
+        If the selected `method` is one of 'krogh', 'piecewise_polynomial',
+        'spline', 'pchip', 'akima':
+        They are wrappers around the scipy interpolation methods of similar
+        names. These use the actual numerical values of the index.
+        For more information on their behavior, see the
+        `scipy documentation
+        <http://docs.scipy.org/doc/scipy/reference/interpolate.html#univariate-interpolation>`__
+        and `tutorial documentation
+        <http://docs.scipy.org/doc/scipy/reference/tutorial/interpolate.html>`__.
 
         Examples
         --------
 
-        Filling in NaNs
+        Filling in `NaN` in a :class:`~pandas.Series` via linear
+        interpolation.
 
         >>> s = pd.Series([0, 1, np.nan, 3])
         >>> s.interpolate()
-        0    0
-        1    1
-        2    2
-        3    3
+        0    0.0
+        1    1.0
+        2    2.0
+        3    3.0
+        dtype: float64
+
+        Filling in `NaN` in a Series by padding, but filling at most two
+        consecutive `NaN` at a time.
+
+        >>> s = pd.Series([np.nan, "single_one", np.nan,
+        ...                  "fill_two_more", np.nan, np.nan, np.nan,
+        ...                  4.71, np.nan])
+        >>> s
+        0              NaN
+        1       single_one
+        2              NaN
+        3    fill_two_more
+        4              NaN
+        5              NaN
+        6              NaN
+        7             4.71
+        8              NaN
+        dtype: object
+        >>> s.interpolate(method='pad', limit=2)
+        0              NaN
+        1       single_one
+        2       single_one
+        3    fill_two_more
+        4    fill_two_more
+        5    fill_two_more
+        6              NaN
+        7             4.71
+        8             4.71
+        dtype: object
+
+        Filling in `NaN` in a Series via polynomial interpolation or splines:
+        Both `polynomial` and `spline` methods require that you also specify
+        an `order` (int).
+
+        >>> s = pd.Series([0, 2, np.nan, 8])
+        >>> s.interpolate(method='polynomial', order=1)
+        0    0.0
+        1    2.0
+        2    5.0
+        3    8.0
         dtype: float64
+        >>> s.interpolate(method='polynomial', order=2)
+        0    0.000000
+        1    2.000000
+        2    4.666667
+        3    8.000000
+        dtype: float64
+
+        Create a :class:`~pandas.DataFrame` with missing values to fill it
+        with diffferent methods.
 
+        >>> df = pd.DataFrame([[0,1,2,0,4],[1,2,3,-1,8],
```

Comment:
Thought I had this comment before but just use the NA values in your constructor - no reason to instantiate the DataFrame with values and then assign them missing values after the fact.

Also make sure you put a space after every comma

Reply:
I changed it so one can see how the columns get created - and we have linear values in 3 columns and quadratic on the 4th.